src/server/wesnothd/server.cpp

```diff
@@ -1807,7 +1807,17 @@ void server::handle_player_in_game(socket_ptr socket, simple_wml::document& data
 	} else if(data.child("stop_updates")) {
 		g.send_data(data, socket);
 		return;
-		// Data to ignore.
+	} else if(data.child("game_history_request")) {
+		if(user_handler_) {
+			int offset = data.child("game_history_request")->attr("offset").to_int();
+			std::string player_name = data.child("game_history_request")->attr("search_for").to_string();
+			int player_id = user_handler_->get_forum_id(player_name);
+
+			LOG_SERVER << "Querying game history for player `" << player_name << "`." << std::endl;
+			user_handler_->get_game_history(io_service_, *this, socket, player_id, offset);
+		}
+		return;
```

Comment:
It doesn't seem this result is ever actually sent to player 🤔

Reply:
That happens in the `forum_user_handler`'s method.